container_api/container.py

```diff
@@ -57,13 +78,53 @@ def fetch_data(self, retries=3):
 		return read_container(self.container_id,
 			sort_function=self.sort_function,
 			additional_capi_params=self.additional_capi_params)
+		
+	def fetch_title_override(self):
+		return read_container_title(self.container_id)
 
 
 class FrontDataSource:
 	def __init__(self, front_id, metadata=None):
 		self.front_id = front_id
 		self.metadata = metadata
+	
+	def fetch_title_override(self, retries=3):
+		container_api_host = configuration.read('CONTAINER_API_HOST')
+		try:
+			url = "http://{host}/list/collections/by/front/{front_id}".format(
+				host=container_api_host,
+				front_id=self.front_id)
+
+			if self.metadata:
+				url = "{url}/and/by/metadata/{metadata}".format(
+					url=url,
+					metadata=self.metadata)
+			
+			result = urlfetch.fetch(url, deadline=9)
+			
```

Comment:
I think it might be useful to separate L92-103 into a separate method to explain what this part does? If I remember fronts tool correctly is this fetching container information for a front? Perhaps that could be separate method called `fetchContainerForFront`?